### PR TITLE
Run fireperf e2e test from the public reepo.

### PR DESCRIPTION
**Pull request by yifanyang**
Moving fireci plugin for fireperf end to end test from the release repo to the public repo.

### CODE REVIEW

File: ci/fireci/fireciplugins/fireperf.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+def _find_fireperf_plugin_version():
+  local_maven_repo_dir = pathlib.Path.home().joinpath('.m2', 'repository')
+  artifacts_path = local_maven_repo_dir.joinpath('com', 'google', 'firebase', 'perf-plugin')
+  versions = [pom.parent.name for pom in artifacts_path.rglob('*.pom')]
+  snapshots = list(filter(lambda v: v.endswith('-SNAPSHOT'), versions))
+
+  if snapshots:
+    return snapshots[0]
+  else:
+    raise click.ClickException('Cannot find a snapshot version in local maven repo.')
```
Comment (ramanpreetSinghKhinda) [inline ci/fireci/fireciplugins/fireperf.py:85]: I assume every run will be a fresh and clean run so we'll only have one version to pick which is the latest HEAD, right?

**Review by ramanpreetSinghKhinda — APPROVED** [review]

File: ci/fireci/fireciplugins/fireperf.py (hunk 1 above)
Comment (yifanyang) [reply ci/fireci/fireciplugins/fireperf.py:85]: Yes, exactly. There will be only one or none, depending on whether `:firebase-performance:performance-gradle:publishToMavenLocal` succeeds.

**Review by jeremyjiang-dev — APPROVED** [review]
The next step will be removing the same file in the `FirebasePrivate
/firebase-android-release` repo?

**Comment by visumickey** [comment]
@yifanyang Thanks for making those changes. There are a couple of things I would like to learn a bit about:

1. How can we learn about errors when running the test? Eg. if there are any changes to the SDK that is not compatible with the plugin, it would good to learn them.
2. Where would we configure this script to run the E2E test app? Are there any other config changes required to enable this app?

**Comment by yifanyang** [comment]
> The next step will be removing the same file in the `FirebasePrivate /firebase-android-release` repo?

Yes. It will be done in this [PR](https://github.com/FirebasePrivate/firebase-android-release/pull/834). I'll submit them together.

**Comment by yifanyang** [comment]
> @yifanyang Thanks for making those changes. There are a couple of things I would like to learn a bit about:
> 
> 1. How can we learn about errors when running the test? Eg. if there are any changes to the SDK that is not compatible with the plugin, it would good to learn them.
> 2. Where would we configure this script to run the E2E test app? Are there any other config changes required to enable this e2e test app run?

The status of recent runs for e2e test can be found here:
* (autopush) https://prow-gob.gcpnode.com/job-history/gs/fireescape/logs/periodic-fireperf-e2e-test-autopush/
* (prod) https://prow-gob.gcpnode.com/job-history/gs/fireescape/logs/periodic-fireperf-e2e-test-prod/

We have also set up an alert for these two jobs, which sends out emails in case of failures. @vkryachko Can you add fireperf team as a recipient of the alert?

This script is actually invoked/configured [here](https://team.git.corp.google.com/firebase-android-core/test-infra/+/refs/heads/master/prow/config.yaml) (search `fireperf-e2e-test` in the file). The other config changes required for this migration is https://team-review.git.corp.google.com/c/firebase-android-core/test-infra/+/966612. I saw Raman has already added you as a reviewer though.

**Pull request merged**